server/pulp/server/async/worker_watcher.py

```diff
@@ -78,9 +78,6 @@ def handle_worker_heartbeat(event):
         msg = _("New worker '%(worker_name)s' discovered") % event_info
```

Comment:
I had imagined this entire functionality being distributed to where the heartbeats are recorded. That would also eliminate the need to have this thread altogether. Does that sound right?